datafusion/functions-nested/src/min.rs

```diff
@@ -0,0 +1,127 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::utils::make_scalar_function;
```

Comment:
I recommend putting this in the same file as  https://github.com/apache/datafusion/blob/main/datafusion/functions-nested/src/max.rs

perhaps call it https://github.com/apache/datafusion/blob/main/datafusion/functions-nested/src/min_max.rs 🤔 

You may be able to share some code, but more importantly I think it will be clearer that min and max are basically the same function